pkg/agents/team4/finalAgent/trustScoreUpdates.go

```diff
@@ -0,0 +1,117 @@
+package team4EvoAgent
+
+import (
+	"fmt"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
+)
+
+func (a *CustomAgentEvo) CheckForResponse(msg messages.BoolResponseMessage) {
+	if a.PlatformOnFloor() && len(a.params.responseMessages) > 0 { // Check if there are any responses messages.
+		for i := 0; i < len(a.params.responseMessages); i++ { // Iterate through each response message
```

Comment:
```suggestion
		for index, msg := range a.params.responseMessages { // Iterate through each response message
```
This means you don't need to index things. You can use `_` if you don't need either index or msg.